### PR TITLE
Add auth helpers

### PR DESCRIPTION
**Pull request by turkenh**
Signed-off-by: Hasan Turken <turkenh@gmail.com>

### Description of your changes

Fixes "no Auth Provider found" errors like mentioned in https://github.com/crossplane-contrib/provider-helm/issues/84#issuecomment-818871124

This could also be related with #72 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

With a helm release pointing a GKE cluster in provider-config

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -20,12 +20,12 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
```
Comment (hasheddan) [inline cmd/provider/main.go:23]: nitpick: I don’t think this should be moved

File: go.mod [hunk 2]
```diff
@@ -22,3 +22,5 @@ require (
 	sigs.k8s.io/kustomize/api v0.5.1
 	sigs.k8s.io/yaml v1.2.0
 )
+
+replace github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.3.2+incompatible
```
Comment (hasheddan) [inline go.mod:26]: Any idea why this is required?

Comment (turkenh) [reply go.mod:26]: I added this to resolve following dependency errors:

```
github.com/crossplane-contrib/provider-helm/cmd/provider imports
	k8s.io/client-go/plugin/pkg/client/auth imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest: ambiguous import: found package github.com/Azure/go-autorest/autorest in multiple modules:
	github.com/Azure/go-autorest v10.8.1+incompatible (/Users/hasanturken/Go/pkg/mod/github.com/!azure/go-autorest@v10.8.1+incompatible/autorest)
	github.com/Azure/go-autorest/autorest v0.9.0 (/Users/hasanturken/Go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.0)
github.com/crossplane-contrib/provider-helm/cmd/provider imports
	k8s.io/client-go/plugin/pkg/client/auth imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest/adal: ambiguous import: found package github.com/Azure/go-autorest/autorest/adal in multiple modules:
	github.com/Azure/go-autorest v10.8.1+incompatible (/Users/hasanturken/Go/pkg/mod/github.com/!azure/go-autorest@v10.8.1+incompatible/autorest/adal)
	github.com/Azure/go-autorest/autorest/adal v0.5.0 (/Users/hasanturken/Go/pkg/mod/github.com/!azure/go-autorest/autorest/adal@v0.5.0)
github.com/crossplane-contrib/provider-helm/cmd/provider imports
	k8s.io/client-go/plugin/pkg/client/auth imports
	k8s.io/client-go/plugin/pkg/client/auth/azure imports
	github.com/Azure/go-autorest/autorest/azure: ambiguous import: found package github.com/Azure/go-autorest/autorest/azure in multiple modules:
	github.com/Azure/go-autorest v10.8.1+incompatible (/Users/hasanturken/Go/pkg/mod/github.com/!azure/go-autorest@v10.8.1+incompatible/autorest/azure)
	github.com/Azure/go-autorest/autorest v0.9.0 (/Users/hasanturken/Go/pkg/mod/github.com/!azure/go-autorest/autorest@v0.9.0/azure)
```

Comment (hasheddan) [reply go.mod:26]: Oof gotta love k8s imports 🙃

File: cmd/provider/main.go (hunk 1 above)
Comment (turkenh) [reply cmd/provider/main.go:23]: grouping non-repo ones together, also consistent with the rest of the repo as far as I could see.

Comment (hasheddan) [reply cmd/provider/main.go:23]: Typically we group org related ones together, but I guess that this one is _technically_ in a different org 👍

**Review by hasheddan — APPROVED** [review]

**Pull request merged**